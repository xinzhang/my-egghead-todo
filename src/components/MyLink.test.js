import React from 'react';
import MyLink from './MyLink';
import renderer from 'react-test-renderer';

test('Link change the class when hovered', () => {
  const component = renderer.create(
    <MyLink page="http://www.google.com">Google</MyLink>
  );

  let tree = component.toJSON();
  expect(tree).toMatchSnapshot();

  // manually trigger the callback
    tree.props.onMouseEnter();
    // re-rendering
    tree = component.toJSON();
    expect(tree).toMatchSnapshot();

    // manually trigger the callback
      tree.props.onMouseLeave();
      // re-rendering
      tree = component.toJSON();
      expect(tree).toMatchSnapshot();
})
