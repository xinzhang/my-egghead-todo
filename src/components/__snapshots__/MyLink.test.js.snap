exports[`test Link change the class when hovered 1`] = `
<a
  className="normal"
  href="http://www.google.com"
  onMouseEnter={[Function]}
  onMouseLeave={[Function]}>
  Google
</a>
`;

exports[`test Link change the class when hovered 2`] = `
<a
  className="hovered"
  href="http://www.google.com"
  onMouseEnter={[Function]}
  onMouseLeave={[Function]}>
  Google
</a>
`;

exports[`test Link change the class when hovered 3`] = `
<a
  className="normal"
  href="http://www.google.com"
  onMouseEnter={[Function]}
  onMouseLeave={[Function]}>
  Google
</a>
`;
